receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -129,6 +188,35 @@ func parseMessageToMetric(line string) (*statsDMetric, error) {
 			return nil, fmt.Errorf("unrecognized message part: %s", part)
 		}
 	}
+
+	switch result.statsdMetricType {
+	case "g":
+		f, err := strconv.ParseFloat(result.value, 64)
+		if err != nil {
+			return nil, fmt.Errorf("gauge: parse metric value string: %s", result.value)
+		}
+		result.floatvalue = f
+		result.metricType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+	case "c":
+		f, err := strconv.ParseFloat(result.value, 64)
+		if err != nil {
+			return nil, fmt.Errorf("counter: parse metric value string: %s", result.value)
+		}
+		i := int64(f)
+		if 0 < result.sampleRate && result.sampleRate < 1 {
+			i = int64(f / result.sampleRate)
```

Comment:
Sample-rate is not defined for Gauge ("g") in statsd. It is defined for Counter ("c"), Timing ("ms"), and Histogram ("h").

Reply:
Yes, here is for Counter. I do not take sample rate into consideration for Gauge.